service-config/src/main/java/com/palantir/remoting2/config/service/ProxyConfiguration.java

```diff
@@ -36,45 +36,105 @@
 @Style(visibility = Style.ImplementationVisibility.PACKAGE, builder = "new")
 public abstract class ProxyConfiguration {
 
+    enum Type {
+
+        /**
+         * Use a direct connection. This option will bypass any JVM-level configured proxy settings.
+         */
+        direct,
```

Comment:
SslConfiguration uses UPPER_CASE, i think. What are example of lower-case configs?